lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -205,15 +205,15 @@ public ColumnReader create(PrimitiveField field, AggregatedMemoryContext aggrega
             }
             if (timestampType.isShort()) {
                 return switch (timestampAnnotation.getUnit()) {
-                    case MILLIS -> createColumnReader(field, valueDecoders::getInt64TimestampMillsToShortTimestampDecoder, LONG_ADAPTER, memoryContext);
-                    case MICROS -> createColumnReader(field, valueDecoders::getInt64TimestampMicrosToShortTimestampDecoder, LONG_ADAPTER, memoryContext);
-                    case NANOS -> createColumnReader(field, valueDecoders::getInt64TimestampNanosToShortTimestampDecoder, LONG_ADAPTER, memoryContext);
+                    case MILLIS -> createColumnReader(field, (encoding) -> valueDecoders.getInt64TimestampMillsToShortTimestampDecoder(encoding, timeZone), LONG_ADAPTER, memoryContext);
```

Comment:
```suggestion
case MILLIS -> createColumnReader(field, encoding -> valueDecoders.getInt64TimestampMillsToShortTimestampDecoder(encoding, timeZone), LONG_ADAPTER, memoryContext);
```

same for the other `(encoding) ->` occurences